src/ray/raylet/scheduling/scheduler_resource_reporter.cc

```diff
@@ -130,8 +134,21 @@ void SchedulerResourceReporter::FillResourceUsage(
 
   fill_resource_usage_helper(
       tasks_to_schedule_ | boost::adaptors::transformed(transform_func), false);
-  fill_resource_usage_helper(
-      tasks_to_dispatch_ | boost::adaptors::transformed(transform_func), false);
+
+  auto tasks_to_dispatch_range =
+      tasks_to_dispatch_ | boost::adaptors::transformed([](const auto &pair) {
+        auto cnt = pair.second.size();
+        // We should only report dispatching tasks that do not have resources allocated.
+        for (const auto &task : pair.second) {
```

Comment:
The O(n) loop is probably ok since the codepath is triggered outside of the critical path of tasks running. 

Will do core benchamrk to validate.

Reply:
what's the frequency? I think it is hard to validate it via benchmark because it will affect the CPU usage or the latency to reply some RPCs (and probably e2e latency is not as affected)

Reply:
right now, 1 every sec.

Reply:
That seems okay then!